File: Data Structures & Algorithms/string-encode-and-decode/submission-1.swift
class Solution {

    func encode(_ strs: [String]) -> String {
         var result = ""

         for str in strs {
            result.append(str + "-")
         }

         return result
    }

    func decode(_ str: String) -> [String] {
        var result: [String] = str.components(separatedBy: "-").dropLast()
        return result
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int result = 0;

        for (auto c : s) {
            int count = 0, l = 0;
            for (int r = 0; r < s.size(); r++)  {
                if (s[r] == c) {
                    count++;
                }

                while ((r - l + 1) - count > k) {
                    if (s[l] == c) {
                        count --;
                    }
                    l++;
                }

                result = max(result, r - l + 1);
            }
        }
        return result;
    }
};
